src/prepare_problem.py

```diff
@@ -0,0 +1,123 @@
+"""This work was partially funded by the EuroHPC PL project,
+funded in the frame of Smart Growth Operational Programme, topic 4.2.
+
+Authors: Mateusz Hurbol, Justyna Zawalska
+"""
+
+import inspect
+import math
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Dict, List, Tuple
+
+import networkx as nx
+import pandas as pd
+import wfcommons
+from dataclasses_json import dataclass_json
+
+from src.utils.file_management import read_json_file
+
+
+@dataclass_json
+@dataclass
+class Task:
+    """
+    Class with the properties of a task.
+    """
+    id: str
+    name: str
+    machine: str
+    memory: float
+    runtime: float
+
+    @classmethod
+    def from_dict(cls, env):
+        """Allows to extract only selected fields that are needed to create the Task class."""
+
+        return cls(**{
+            k: v for k, v in env.items()
+            if k in inspect.signature(cls).parameters
+        })
+
+
+def get_machines_from_wfcommons_file(workflow_filename: str) -> Dict[str, dict]:
```

Comment:
Please use everywhere `Dict` if you are below python 3.9
```suggestion
def get_machines_from_wfcommons_file(workflow_filename: str) -> Dict[str, Dict]:
```